src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,44 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor() {
+    super();
+    this.hidden = false;
+
+    Herbivore.alive.push(this);
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor() {
+    super();
+
+    Carnivore.alive.push(this);
```

Comment:
[CODE PERFOMANCE] - Instead of pushing the new instance to `Carnivore.alive`, you should add it to `Animal.alive`. This will ensure that all alive animals are tracked in one array, which is the intended design according to the task description.